libs/common-utils/src/getChainIdImmediately.ts

```diff
@@ -10,3 +15,7 @@ export async function getChainIdImmediately(provider: JsonRpcProvider): Promise<
     return undefined
   }
 }
+
+function isJsonRpcProvider(provider: JsonRpcProvider | Provider): provider is JsonRpcProvider {
+  return (provider as JsonRpcProvider).send !== undefined
```

Comment:
good enough, im happier now :) 

wouldn't `instanceof JsonRpcProvider` be a good idea?

Reply:
`instanceof` works for when it already is a JsonRpcProvider
Checking for the `send` method works for both